cores/esp32/Print.cpp

```diff
@@ -52,19 +52,24 @@ size_t Print::printf(const char *format, ...)
     va_list copy;
     va_start(arg, format);
     va_copy(copy, arg);
-    size_t len = vsnprintf(NULL, 0, format, copy);
+    int len = vsnprintf(temp, sizeof(loc_buf), format, copy);
     va_end(copy);
+    if(len < 0) {
+        va_end(arg);
+        return 0;
+    };
     if(len >= sizeof(loc_buf)){
-        temp = new char[len+1];
+        temp = (char*) malloc(len+1);
         if(temp == NULL) {
+            va_end(arg);
             return 0;
         }
+        len = vsnprintf(temp, len+1, format, arg);
     }
-    len = vsnprintf(temp, len+1, format, arg);
     write((uint8_t*)temp, len);
```

Comment:
I've just thought of something, This print function is used for many different subsystems. I think some of them may actually have cases where `write()` can fail. instead of returning the number of characters to be written, actually return the true number.
```c++
int actualLen = write((uint8_t*)temp,len);
...
return actualLen;
```

Reply:
Ill add/fix that